xtask/src/main.rs

```diff
@@ -0,0 +1,95 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+type DynError = Box<dyn std::error::Error>;
+
+fn main() {
+    if let Err(e) = try_main() {
+        eprintln!("{}", e);
+        std::process::exit(-1);
+    }
+}
+
+fn try_main() -> Result<(), DynError> {
+    let task = env::args().nth(1);
+    match task.as_deref() {
+        Some("build") => build()?,
+        _ => print_help(),
+    }
+    Ok(())
+}
+
+fn print_help() {
+    eprintln!(
+        "Tasks:
+
+wasm            builds wasm using wasm32-wasi target
+components      builds components using wasm-tools
+"
+    )
+}
+
+fn build() -> Result<(), DynError> {
+    let _ = fs::remove_dir_all(&dist_dir());
+    fs::create_dir_all(&dist_dir())?;
+
+    build_wasm()?;
+    build_component_function()?;
+
+    Ok(())
+}
+
+fn build_wasm() -> Result<(), DynError> {
+    let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+    let status = Command::new(cargo)
+        .current_dir(project_root())
+        .args(&[
+            "build",
+            "--target=wasm32-wasi",
+            "--release",
+            "--package=function",
+        ])
+        .status()?;
+
+    if !status.success() {
+        Err("`cargo build --target wasm32-wasi --release --package function` failed")?;
+    }
+
+    Ok(())
+}
+
+fn build_component_function() -> Result<(), DynError> {
+    // TODO: unsure about env::var("WASM_TOOLS")
+    let cargo = env::var("WASM_TOOLS").unwrap_or_else(|_| "wasm-tools".to_string());
+    let status = Command::new(cargo)
+        .current_dir(project_root())
+        .args(&[
+            "component",
+            "new",
+            "./target/wasm32-wasi/release/function.wasm",
```

Comment:
I wonder if we can use the `project_root` fn to build this using an absolute path. I've had some issues in the past with path refs across different platforms